prow/cmd/pipeline/controller.go

```diff
@@ -435,11 +445,15 @@ func reconcile(c reconciler, key string) error {
 			return fmt.Errorf("error preparing resources: %v", err)
 		}
 
-		// Create the any git resources that are needed.
+		var pipelineResources []*pipelinev1alpha1.PipelineResource
+
+		// Create the git resources that are needed.
 		for _, res := range resources {
 			resKey := toKey(ctx, namespace, res.Name)
 			logrus.Infof("Create PipelineResource/%s", resKey)
-			if _, err = c.createPipelineResource(ctx, namespace, &res); err != nil {
+			if pr, err := c.createPipelineResource(ctx, namespace, &res); err != nil {
+				pipelineResources = append(pipelineResources, pr)
```

Comment:
Why do we do the append only when we had an error? Also, its not safe to use other return args when err was not nil

Reply:
Fixed